keras_nlp/tokenizers/sentence_piece_tokenizer.py

```diff
@@ -121,6 +126,26 @@ def __init__(
 
         super().__init__(dtype=dtype, **kwargs)
 
+        self.proto = None
+        self.sequence_length = sequence_length
+        self.set_vocabulary(proto)
+
+    def save_assets(self, dir_path):
+        path = os.path.join(dir_path, VOCAB_FILENAME)
+        with open(path, "w") as file:
+            for token in self.proto:
+                file.write(f"{token}\n")
+
+    def load_assets(self, dir_path):
+        path = os.path.join(dir_path, VOCAB_FILENAME)
+        self.set_vocabulary(path)
+
+    def set_vocabulary(self, proto):
```

Comment:
I think we should just call this `set_proto`. A little weird, but matches the init arg naming we have today. And this should not be set on the results of `get_vocabulary`, that would blow up.

```
word_piece_tokenizer.set_vocabulary
sentence_piece_tokenizer.set_proto
byte_pair_tokenizer.set_vocabulary_and_merges
```

Reply:
Got it, I've switched them to `set_proto` throughout. Thanks!